### PR TITLE
[3.2-RC] Document Blazor `loadBootResource`

### LINKED ISSUE #18031: [3.2-rc] Document Blazor `loadBootResource` configuration option
Recently, a `loadBootResource` option was added to Blazor's JS API ([link](https://github.com/dotnet/aspnetcore/pull/20903)) that allows users to configure where different app resources should be loaded from.

We should add some documentation (perhaps under the Host + Deploy) section that explains to users how this API can be used.

```js
Blazor.start({
  loadBootResource: function (type, name, defaultUri, integrity) {
    if (type === "timezonedata") {
      return "https://my-awesome-cdn.com/" + name;    
    }
  },
});
````

Some scenarios that this API can be used for:
- Allow users to load static resources (like timezone data or dotnet.wasm) from CDN
- Loading compressed assemblies using a HTTP request and decompressing them on the client for hosts that don't support fetching compressed contents from the server
- Aliasing resources to a different name by redirecting each fetch request to the new name

cc: @javiercn @SteveSandersonMS

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #18031

[Internal Review Topic (links to section)](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/blazor/webassembly?view=aspnetcore-3.1&branch=pr-en-us-18049#custom-boot-resource-loading)

Safia, the draft text is a hybrid of what you and Steve provided. Also, I put you on the author byline for providing some of the text. Let me know if you prefer a different URL for the link.

cc: @Lupusa87 ... in case you would like to look this over and see if it gives you all of the information that you need for the scenario that you asked about on https://github.com/dotnet/aspnetcore/issues/21042.

### CODE REVIEW

**Comment by Lupusa87** [comment]
@guardex thank you, it looks great, exactly what I was asking.
Simple to use and well done.

File: aspnetcore/host-and-deploy/blazor/webassembly.md [hunk 1]
```diff
@@ -325,3 +325,59 @@ The `--urls` argument sets the IP addresses or host addresses with ports and pro
...
+| `integrity`  | The integrity string representing the expected content in the response. |
+
+`loadBootResource` returns any of the following to override the loading process:
+
+* URI string. Example: `https://my-awesome-cdn.com/blazorwebassembly/3.2.0/${name}`
+* `Promise<Response>`. Example: `return fetch(someUrl);`
+* `null`/`undefined`, which results in the default loading behavior.
+
+External sources must return the required CORS headers for browsers to allow the cross-origin resource loading. CDNs usually provide the required headers by default.
+
```
Comment (captainsafia) [inline aspnetcore/host-and-deploy/blazor/webassembly.md:353]: I'd also call out that if users are issuing their own fetch requests for resources in the `loadBootJson` then they should pass the integrity property in the header.

Comment (guardrex) [reply aspnetcore/host-and-deploy/blazor/webassembly.md:353]: `loadBootJson` isn't described here (or anywhere in the docs), so it needs more explanation.

Comment (danroth27) [reply aspnetcore/host-and-deploy/blazor/webassembly.md:353]: @guardrex I think @captainsafia was still referring to `loadBootResource`. If you make your fetch requests in `loadBootJson` then you need to pass the integrity value properly. See the example I added to the RC announcement post for this: https://devblogs.microsoft.com/aspnet/blazor-webassembly-3-2-0-release-candidate-now-available/

Comment (guardrex) [reply aspnetcore/host-and-deploy/blazor/webassembly.md:353]: I see ... I'll address that now here.

**Comment by guardrex** [comment]
@captainsafia Try now ... I re-organized it a bit. We have two examples now, which each matched a bullet point. It's probably best if they go under their respective bullets now.

File: aspnetcore/host-and-deploy/blazor/webassembly.md [hunk 2]
```diff
@@ -325,3 +325,72 @@ The `--urls` argument sets the IP addresses or host addresses with ports and pro
 ## Configure the Linker
 
 Blazor performs Intermediate Language (IL) linking on each Release build to remove unnecessary IL from the output assemblies. For more information, see <xref:host-and-deploy/blazor/configure-linker>.
+
+## Custom boot resource loading
+
+A Blazor WebAssembly app can use `loadBootResource` to override the built-in boot resource loading mechanism. Use `loadBootResource` for the following scenarios:
```
Comment (captainsafia) [inline aspnetcore/host-and-deploy/blazor/webassembly.md:331]:
```suggestion
A Blazor WebAssembly app can be initialized with the `loadBootResource` function to override the built-in boot resource loading mechanism. Use `loadBootResource` for the following scenarios:
```

**Pull request merged**